### PR TITLE
introduce typeText helper

### PR DESCRIPTION
**Pull request by lstrzebinczyk**
This MR introduces `typeText` helper, as discussed in https://github.com/cibernox/ember-native-dom-helpers/issues/123.

@cibernox What do you think?

### CODE REVIEW

**Comment by cibernox** [comment]
I think that the order of those events is wrong. Check https://ember-twiddle.com/e1b4942ce1ec05e5914c98a93ed13236?openFiles=controllers.application.js%2C

Differences I see when typing on the twiddle:
- I think that only one `change` event should be fired at the end of the of the interaction, because the event only fires when the user focuses out the input field (we can assume that they will unfocus at some point)
- You are missing the `input` event between keydown and up. The order of the events should be <fire keydown>, <set new value>, <fire input>, <fire keyup>, (repeat N times) and finally<fire change>

Another thing that is not wrong but I believe I would do differently is to reuse the public `keyEvent` helper inside this one, so you don't have to wrap everything you do in `run`

**Comment by KillaPL** [comment]
Good catches. I reorganised the events flow and used `keyEvent`. I'm happy to introduce further changes to make the code fit your standards.

File: addon-test-support/type-text.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  @return {RSVP.Promise}
+  @public
+*/
+export function typeText(selector, text) {
+  let el = getElementWithAssert(selector);
+
+  run(() => focus(el));
+  keyEvent(el, 'keydown', BACKSPACE_CODE);
+  run(() => el.value = '');
+  keyEvent(el, 'input');
```
Comment (cibernox) [inline addon-test-support/type-text.js]: `input` is not a key event, that one must be fired with `run(() => fireEvent(el, 'input'));`

I'm also thinking that probably I wouldn't bake in the backspace simulation in the helper. I think that user could start typing right away.

File: tests/integration/type-text-test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    } else {
+      assert.equal(String.fromCharCode(keyCode), message[onKeyUpIndex]);
+    }
+    onKeyDownIndex += 1;
+  };
+
+  let onInputIndex = 0;
+  this.onInput = (e) => {
+    if (onInputIndex === 0) {
+      assert.equal(e.target.value, '');
```
Comment (cibernox) [inline tests/integration/type-text-test.js]: If you remove the backspace, you wouldn't need this too.

File: addon-test-support/type-text.js (hunk 1 above)
Comment (KillaPL) [reply addon-test-support/type-text.js]: Oh boy, that's what you get when you try to get something done on friday evening... : ( Will be done shortly.

Comment (KillaPL) [reply addon-test-support/type-text.js]: Actually, how should removing current input be handled? There might be already something in the input, and  helper probably should overwrite it, not append to it. Should it simply clean the value, but  not fire any events?

Comment (KillaPL) [reply addon-test-support/type-text.js]: @cibernox I suppose I should have mentioned you.

**Comment by lstrzebinczyk** [comment]
I'm gonna close this one, it's been ages.

**Pull request closed without merging**